bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/SpeedtestHandler.java

```diff
@@ -286,18 +307,70 @@ private void getSpeed() {
+                downloadBytes = new QuantityType<>(Double.parseDouble(tmpCont.getDownload().getBytes()), Units.BYTE);
+                downloadElapsed = new QuantityType<>(Double.parseDouble(tmpCont.getDownload().getElapsed()) / 1000.0,
+                        Units.SECOND);
+                uploadBandwidth = new QuantityType<>(Double.parseDouble(tmpCont.getUpload().getBandwidth()) / 125000.0,
+                        Units.MEGABIT_PER_SECOND);
+                uploadBytes = new QuantityType<>(Double.parseDouble(tmpCont.getUpload().getBytes()), Units.BYTE);
+                uploadElapsed = new QuantityType<>(Double.parseDouble(tmpCont.getUpload().getElapsed()) / 1000.0,
+                        Units.SECOND);
                 isp = tmpCont.getIsp();
                 interfaceInternalIp = tmpCont.getInterface().getInternalIp();
                 interfaceExternalIp = tmpCont.getInterface().getExternalIp();
                 resultUrl = tmpCont.getResult().getUrl();
+
+                HttpClient client = httpClient;
+                if (client == null) {
+                    logger.debug("Unable to download image because httpClient is not set");
+                } else {
+                    try {
+                        String url = String.valueOf(resultUrl) + ".png";
+                        logger.debug("Downloading result image from: {}", url);
+                        Request request = client.newRequest(url);
+                        request.method(HttpMethod.GET);
+                        request.timeout(API_REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+                        ContentResponse contentResponse = request.send();
+                        switch (contentResponse.getStatus()) {
+                            case HttpStatus.OK_200:
+                                resultImage = new RawType(contentResponse.getContent(),
+                                        contentResponse.getHeaders().get(HttpHeader.CONTENT_TYPE));
+                                break;
+                            default:
+                                logger.debug("HTTP GET failed: {}, {}", contentResponse.getStatus(),
+                                        contentResponse.getReason());
+                                break;
+                        }
+                    } catch (TimeoutException e) {
+                        resultImage = UnDefType.NULL;
+                        logger.debug("TimeoutException: Download of result image timed out");
+                    } catch (ExecutionException e) {
+                        resultImage = UnDefType.NULL;
+                        logger.debug("ExecutionException: {}", e.getMessage());
+                    } catch (InterruptedException e) {
+                        resultImage = UnDefType.NULL;
+                        logger.debug("InterruptedException: {}", e.getMessage());
+                        Thread.currentThread().interrupt();
+                    }
```

Comment:
Look at the method HttpUtil.downloadImage in core framework, it should match your need and avoid that your rewrite code.

Reply:
I mean this method:
https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/HttpUtil.java#L362

Look at in the same class, you have variants with more parameters in case you would like to control the timeout for example.

Reply:
thanks for the hint!